doc/contributing/adding-v8-fast-api.md

```diff
@@ -0,0 +1,153 @@
+# Adding V8 Fast API
+
+Node.js uses [V8](https://github.com/v8/v8) as the JavaScript engine.
+In order to provide fast paths for functions that are called quite often,
+V8 offers the usage of `fast API` functions that does not trigger JavaScript
+functions or allocate heap on JavaScript.
+
+## Limitations
+
+* Fast API functions can not use JavaScript heap allocation.
+* Fast API functions can not trigger JavaScript execution.
+* Reporting/returning errors is not available on fast API, but can be done
+  through fallbacks to slow API.
+* Not all parameter and return types are supported in fast API calls.
+  For a full list, please look into
+  [`v8-fast-api-calls.h`](../../deps/v8/include/v8-fast-api-calls.h) file.
+
+## Requirements
+
+* Each unique fast API function signature should be defined inside the
+  [`external references`](../../src/node_external_reference.h) file.
+* To test fast APIs, make sure to run the tests in a loop with a decent
+  iterations count to trigger V8 for optimization and to prefer the fast API
+  over slow one.
+
+## Fallback to slow path
+
+Fast API supports fallback to slow path in case logically it is wise to do so,
+for example when providing a detailed error, or need to trigger JavaScript.
+Fallback mechanism can be enabled and changed from both the caller
+JavaScript function or from the fast API function declaration.
+
+Every fast API function accessible from JavaScript side can pass an object
+consisting of fallback key with a boolean value as the last parameter
+(or the first parameter, if no parameters of the function exist).
+
+Note that, the fallback options parameter does not need to be defined
+in the C++ function in order for it to be used in the JavaScript side.
+
+In V8, the options fallback is defined as `FastApiCallbackOptions` inside
+[`v8-fast-api-calls.h`](../../deps/v8/include/v8-fast-api-calls.h).
```

Comment:
Referring to my previous comment, I don't think we should give an exact line in here, since it will change in time, and it would be hard to keep track of it.